src/middleware/serverMiddlewareLoader.middleware.ts

```diff
@@ -4,32 +4,39 @@ import bodyParser from 'body-parser'
 import { createLoggerMiddleware } from 'ch-logging'
 import ApplicationLogger from 'ch-logging/lib/ApplicationLogger'
 import cookieParser from 'cookie-parser'
-import * as express from 'express'
+import { Application, NextFunction, Request, RequestHandler, Response } from 'express'
 import { INTERNAL_SERVER_ERROR } from 'http-status-codes'
 import { inject } from 'inversify'
 import { provide } from 'inversify-binding-decorators'
 import NunjucksLoader from './nunjucksLoader.middleware'
 
 import { APP_NAME } from 'app/constants/app.const'
+import TYPES from 'app/types'
 
 @provide(ServerMiddlewareLoader)
 export default class ServerMiddlewareLoader {
 
   public constructor(
     @inject(NunjucksLoader) private nunjucks: NunjucksLoader,
-    @inject(ApplicationLogger) private logger: ApplicationLogger) {}
+    @inject(ApplicationLogger) private logger: ApplicationLogger,
+    @inject(TYPES.SessionMiddleware) private sessionMiddleware: RequestHandler,
+    @inject(TYPES.SaveUserEmailToLocals) private saveUserEmailToLocals: RequestHandler
+  ) {}
 
-  public loadServerMiddleware(app: express.Application, directory: string): void {
+  public loadServerMiddleware(app: Application, directory: string): void {
     app.use(bodyParser.json())
     app.use(bodyParser.urlencoded({ extended: true }))
     app.use(cookieParser())
     app.use(createLoggerMiddleware(APP_NAME))
 
+    app.use(this.sessionMiddleware)
+    app.use(this.saveUserEmailToLocals)
```

Comment:
I wonder in future when we make further changes to this file if we should have a `CustomServerMiddlewareLoader` where we would load middlewares that we or CH have created. I can imagine the number of injects here becoming quite large going forward.

Reply:
Good thought for the future, though currently it is quite a small class